2022-03-monthly-report.org

```diff
@@ -0,0 +1,516 @@
+| Macros and cfg expansion          |        65% |            |       | 1st Dec 2021   | -               | 28th Mar 2022  |
+| Imports and Visibility            |         0% |         0% | -     | 29th Mar 2022  | -               | 27th May 2022  |
+| Const Generics                    |         0% |         0% | -     | 30th May 2022  | -               | 25th Jul 2022  |
+| Intrinsics and builtins           |         0% |         0% | -     | 6th Sept 2021  | -               | 30th Sept 2022 |
+| Borrow checking                   |         0% |         0% | -     | TBD            | -               | TBD            |
+
+*** Risks
+
+| Risk                    | Impact (1-3) | Likelihood (0-10) | Risk (I * L) | Mitigation                                                 |
+|-------------------------+--------------+-------------------+--------------+------------------------------------------------------------|
+| Rust Language Changes   |            3 |                 7 |           21 | Keep up to date with the Rust language on a regular basis  |
+| Going over target dates |            3 |                 5 |           15 | Maintain status reports and issue tracking to stakeholders |
+
+** Planned Activities
+
+
+ 
+** Detailed changelog
+
+*** must use attribute
+
+To support must use, the GCC CPP front-end already supports the C++ nodiscard attribute which is analogus to rust must use attribute. Rust also supports using must use on types which we still need to test/support but this is the building block to support this on functions which discard their results.
+
+#+BEGIN_SRC rust
+#[must_use = "TEST 1"]
+fn test1() -> i32 {
+    123
+}
+
+#[must_use = "TEST 2"]
+fn test2() -> i32 {
+    456
+}
+
+fn main() {
+    let _a = test1();
+
+    test2();
+}
+#+END_SRC
+
+The error respects GCC -Wunused-result but this is turned on by default in the front-end.
+
+#+BEGIN_SRC
+<source>:14:5: warning: ignoring return value of 'example::test2', that must be used: 'TEST 2' [-Wunused-result]
+   14 |     test2();
+      |     ^
+<source>:7:1: note: declared here
+    7 | fn test2() -> i32 {
+      | ^
+#+END_SRC
+
+see: https://godbolt.org/z/81j9G584e
+
+*** Recursive macros using seperators
+
+Macros can be recusive resulting in new macro invocations which need to be expanded. They also can have matchers which are like regular expressions in their matchers which require n-number of arguments delimited by a single matcher to terminate the sequence. This looks very similar to bison grammer files which is pretty impressive how expressive macros are in rust.
+
+#+BEGIN_SRC rust
+macro_rules! add {
+        ($e:expr | $($es:expr) | *) => {
+            $e + add!($($es) | *)
+        };
+        ($e:expr) => {
+            $e
+        };
+    }
+
+fn test() -> i32 {
+    add!(1 | 2 | 3 | 4 | 5 | 6)
+}
+
+#+END_SRC
+
+see: https://godbolt.org/z/TfWrEovf3
+
+*** Implement proper repetition separators
```

Comment:
This section is actually similar to the previous one, I think we should keep only one of the two